inc/application/view/extension/datahelpersextension.class.php

```diff
@@ -55,6 +55,7 @@ public function getFilters(): array {
          new TwigFilter('safe_html', [$this, 'getSafeHtml'], ['is_safe' => ['html']]),
          new TwigFilter('verbatim_value', [$this, 'getVerbatimValue']),
          new TwigFilter('shortcut', [$this, 'underlineShortcutLetter'], ['is_safe' => ['html']]),
+         new TwigFilter('replaceImagesByGallery', [Html::class, 'replaceImagesByGallery']),
```

Comment:
IMHO, it would be preferable to add a new filter `rich_text` with options.
Indeed, `safe_html` would be then just a filter to prevent XSS (could be used to display HTML generated by a tier app), and `rich_text` would be a filter to handle all features related to rich text content in GLPI.
For now, options would be:
- `images_gallery` (bool, false by default),
- `user_mentions` (bool, true by default).

Usage would be `content|rich_text({'images_gallery': true})`

Reply:
What is the purpose of having a `user_mentions` option when the content doesn't seem to need transformed at all for that functionality. I don't really know how that is supposed to work in readonly text blocks. Is it supposed to get transformed to a link to the user, group, etc?

Reply:
Indeed, user mentions are replaced by links, see : https://github.com/glpi-project/glpi/blob/master/inc/features/usermention.class.php#L206

I do not see yet usages where user mentions should not be transformed into links, so I think it should be activated by default. As adding an option to disable this transformation is really simple, I would prefere to add it now.

Reply:
Looking at the code, I figured out that the trait will not be usable in twig context.

Maybe the trait should be transformed into a standalone class, like `Glpi\Content\UserMentions` (do not know what is the best namespace to use), and signature could be:
- `public static function handleUserMentions(CommonDBTM $item): void`
- `public static function refreshUserMentionsHtmlToDisplay(string $content): string` (in this one, the existing `bool $add_link = true` is useless and ca be remove)

Reply:
Glpi\RichText\UserMention makes more sense to me. Keep class name singular like everything else (Cluster, Computer, Ticket, etc). The namespace then matches what was already used for JS script path (js/RichText/UserMention.js).

Reply:
OK for `Glpi\RichText\UserMention`

Reply:
Can I move the existing RichText class from Glpi\Toolbox to the new namespace? I guess then all that is needed is to add a new "applyTransforms" method that does the optional user mention and images transforms, then calls `getSafeHtml`? Or should it be called `getReadonlyContent`?

Reply:
> Can I move the existing RichText class from Glpi\Toolbox to the new namespace?

You can.

> I guess then all that is needed is to add a new "applyTransforms" method that does the optional user mention and images transforms, then calls getSafeHtml? Or should it be called getReadonlyContent?

You can add a `Glpi\RichText\RichText::getEnhancedHtml(array $options)` method (as an english native speaker, you may propose a better name).

If we add this method, then
 - `RichText::getSafeHtml()` should be used only in a rich text edition context
 - `RichText::getEnhancedHtml()` should be used in rendering contexts

That would make a lot of changes, do you need help to replace existing usages of `RichText::getSafeHtml()` ?